### DISCUSSION TITLE
How to decide the running time under using continous source?

### DISCUSSION THREAD

**Discussion by beizhixing** [NanoComp/meep#2806] [General]
Hi, I want to simulate waveguide by using continous source, it is tricky to set the running time. 

The image below are under different simulation time setting:

![E_gama0_15 00](https://github.com/NanoComp/meep/assets/27534638/638f618a-f833-43b4-a48b-4050d9b5e9a6)

sum.run(until=400)

![E_gama0_15 000](https://github.com/NanoComp/meep/assets/27534638/0b0095bc-4150-4461-bc74-f749d95644c8)

sum.run(until=500)

![E_gama0_0 083pi](https://github.com/NanoComp/meep/assets/27534638/921178c9-a184-4fc0-93d2-11d6f9999347)

sum.run(until=3000)

I think as long as the simulation convergence, no matter how long I set, the result shoud be not change. 

I'd be appreciate if you can help me to solve it, @smartalecH @oskooi !

**Comment by stevengj** [comment]
What is the scale bar on the third plot?  Is the simulation diverging?

**Reply by beizhixing** [reply]
This is a 2D simulation. The scale bar are default, 1 micromrter. 
I do not how to judge  whether the simluation converge? But you can see some scattered light when runtime = 3000.

**Reply by stevengj** [reply]
by scale bar I mean the scale bar for the colors, which you omitted in the third plot

**Reply by beizhixing** [reply]
The scale bar of the third same as the second plot.

**Comment by smartalecH** [comment]
Is this just a waveguide? Are you only injecting a single mode?

Do you have proper boundaries layers set up? It looks like you may have some large reflections with the domain edges?

**Reply by beizhixing** [reply]
Yes, it is just a waveguide by anisotropic materials, refer to https://www.nature.com/articles/nphoton.2016.138. I set all the boundary conditions of PML = 1.0um.
